src/cdk-experimental/menu/menu.ts

```diff
@@ -48,10 +56,42 @@ export class CdkMenu extends CdkMenuGroup implements AfterContentInit {
   @ContentChildren(CdkMenuGroup, {descendants: true})
   private readonly _nestedGroups: QueryList<CdkMenuGroup>;
 
+  /**
+   * A reference to the enclosing parent menu panel.
+   *
+   * Required to be set when using ViewEngine since ViewEngine does support injecting a reference to
+   * the parent directive if the parent directive is placed on an `ng-template`. If using Ivy, the
+   * injected value will be used over this one.
+   */
+  @Input('cdkMenuPanel') private readonly _explicitPanel?: CdkMenuPanel;
+
+  constructor(@Optional() private readonly _menuPanel?: CdkMenuPanel) {
+    super();
+  }
+
   ngAfterContentInit() {
     super.ngAfterContentInit();
 
     this._completeChangeEmitter();
+    this._registerWithParentPanel();
+  }
+
+  /** Register this menu with its enclosing parent menu panel */
+  private _registerWithParentPanel() {
+    const parent = this._getMenuPanel();
+    if (parent) {
+      parent._registerMenu(this);
+    } else {
+      throwMissingMenuPanelError();
```

Comment:
This wouldn't strictly be an error- using a menu _outside_ of a panel should be valid (i.e. having an inline menu)

Reply:
Should I remove this now or address it once I implement inline div support?

Reply:
SGTM to do in a follow-up PR